src/compressor/QatAccel.h

```diff
@@ -16,17 +16,25 @@
 #define CEPH_QATACCEL_H
 
 #include <memory>
+#include <mutex>
+#include <condition_variable>
+#include <vector>
 #include <boost/optional.hpp>
 #include "include/buffer.h"
 
-extern "C" struct QzSession_S; //struct QzSession_S comes from QAT libraries
+extern "C" struct QzSession_S; // typedef struct QzSession_S QzSession_T;
 
 struct QzSessionDeleter {
   void operator() (struct QzSession_S *session);
 };
 
 class QatAccel {
-  std::unique_ptr<struct QzSession_S, QzSessionDeleter> session;
+  friend struct cached_session_t;
+  using session_ptr = std::unique_ptr<struct QzSession_S, QzSessionDeleter>;
+  static std::vector<session_ptr> sessions;
```

Comment:
why mark these variables `static`? IIUC, `QatAccel` is a member variable of `Compressor`.  and as a drop-in replacement of popular compressors, `QatAccel` offers the ability to accelerate compressors which are supported, and in general, we either create an instance of compressor on demand. if `QatAccel` is supposed to be shared across different instances of `Compressor`s, probably a better way is to mark `qat_accel` static instead ?

Reply:
Hi, @tchaikov , sorry for replying so late.  I want the initialization of qat to be only once. But direct concurrent writes when RGW starts will result in multiple initializations when static is not used. When initialized multiple times, it will cause RGW to crash.
And I have replaced with `static QatAccel Compressor::qat_accel`.